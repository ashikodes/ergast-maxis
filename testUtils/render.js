import { render } from '@testing-library/react';
import { makeStore } from '../redux/store';

import { RouterContext } from 'next/dist/shared/lib/router-context';
import { Provider } from 'react-redux';

function getWrapper(contexts) {
  function Wrapper({ children }) {
    return (
      <RouterContext.Provider value={contexts.router}>
        <Provider store={contexts.store}>{children}</Provider>
      </RouterContext.Provider>
    );
  }
  return { ...contexts, wrapper: Wrapper };
}

function testRender(
  ui,
  { store = makeStore(), testRouterOptions: router = {} } = {}
) {
  const contexts = { store, router };
  const { wrapper } = getWrapper(contexts);
  return render(ui, { wrapper });
}

export default testRender;
