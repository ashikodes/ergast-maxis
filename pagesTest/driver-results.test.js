import { act } from '@testing-library/react';
import render from '../testUtils/render';
import DriverResultsPage from '../pages/driver-results/[driverId]';

jest.mock('../redux/driver/action');
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn,
  useSelector: jest.fn().mockReturnValue({
    results: [
      {
        raceName: 'brix',
        round: 12,
        Results: [{ position: 6, points: 12 }],
      },
    ],
  }),
}));

describe('Driver results page', () => {
  it('matches snapshot', async () => {
    let container;
    await act(async () => {
      ({ container } = render(<DriverResultsPage />, {
        testRouterOptions: {
          pathname: '/driver-results/max',
          query: { driverId: 'max' },
        },
      }));
    });
    expect(container).toMatchSnapshot();
  });
});
