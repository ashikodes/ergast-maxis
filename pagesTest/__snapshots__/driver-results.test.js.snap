// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Driver results page matches snapshot 1`] = `
<div>
  <div
    class="main-heading"
  >
    <a>
      View overall standings
    </a>
    <h2>
      Driver results
    </h2>
  </div>
  <div
    class="main-content"
  >
    <span
      class="card"
    >
      <h3>
        brix
        <span>
          â†’
        </span>
      </h3>
      <p>
        Position - 
        6
      </p>
      <p>
        Points - 
        12
      </p>
    </span>
  </div>
</div>
`;
